### PR TITLE
Change the eviction metric type and fix rate-limited-timed-queue

### PR DESCRIPTION
**Pull request by gmarek**
People how know better convinced me that aggregate counter is better than a gauge for a number of evictions metric. @Q-Lee 

Per discussion with @pwittrock I add a v1.4 label and a cherrypick candidate label. This is a slightly bigger change than I thought, but it fixes a bug in eviction logic, so it's also important.

cc @derekwaynecarr @smarterclayton @timothysc

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32270)

### CODE REVIEW

File: pkg/controller/node/rate_limited_queue_test.go [hunk 1]
```diff
@@ -303,5 +303,32 @@ func TestSwapLimiter(t *testing.T) {
...
+	evictor.Try(func(value TimedValue) (bool, time.Duration) {
+		deletedMap.Insert(value.Value)
+		return true, 0
+	})
 
+	setPattern := sets.NewString("first", "third")
+	if len(deletedMap) != len(setPattern) {
+		t.Fatalf("Map %v should have length %d", evictor.queue.set, len(setPattern))
+	}
+	if !CheckSetEq(setPattern, deletedMap) {
```
Comment (wojtek-t) [inline pkg/controller/node/rate_limited_queue_test.go]: If you are checking SetEq - then you probably don't need to check for equal lengths in the line above...

File: pkg/controller/node/rate_limited_queue_test.go [hunk 2]
```diff
@@ -303,5 +303,32 @@ func TestSwapLimiter(t *testing.T) {
...
+		t.Errorf("Invalid map. Got %v, expected %v", deletedMap, setPattern)
+	}
+
+	evictor.Add("first", "11111")
+	evictor.Try(func(value TimedValue) (bool, time.Duration) {
+		t.Errorf("We should process the same value if the explicit remove wasn't called.")
```
Comment (wojtek-t) [inline pkg/controller/node/rate_limited_queue_test.go]: s/should/should not/

Comment (gmarek) [inline pkg/controller/node/rate_limited_queue_test.go]: :) Right

File: pkg/controller/node/rate_limited_queue_test.go (hunk 1 above)
Comment (gmarek) [inline pkg/controller/node/rate_limited_queue_test.go]: Yeah - though the distinction was useful for debugging of the original test. I'd rather leave this as is.

**Comment by gmarek** [comment]
@k8s-bot unit test this issue: #IGNORE
federation ingress test bug - @mwielgus is already investigating.

**Comment by wojtek-t** [comment]
lgtm

**Comment by gmarek** [comment]
cc @Q-Lee - it's in.

**Pull request merged**